src/metadata.rs

```diff
@@ -198,7 +199,7 @@ pub(crate) struct Package {
     /// List of dependencies of this particular package
     pub(crate) dependencies: Vec<Dependency>,
     /// Features provided by the crate, mapped to the features required by that feature.
-    pub(crate) features: BTreeSet<String>,
+    pub(crate) features: BTreeMap<String, Vec<String>>,
```

Comment:
(This change does not directly relate to the aim of this PR, but is needed to prepare #81.)